common/Resources/ti.internal/extensions/node/fs.js

```diff
@@ -0,0 +1,1257 @@
+	setTimeout(() => {
+		// flags and mode are optional, we need to handle if not supplied!
+		if (typeof flags === 'function') {
+			callback = flags;
+			flags = 'r';
+			mode = 0o666;
+		} else if (typeof mode === 'function') {
+			callback = mode;
+			mode = 0o666;
+		}
+
+		let fileDescriptor;
+		try {
+			fileDescriptor = fs.openSync(path, flags, mode);
+		} catch (e) {
+			callback(e);
+			return;
+		}
+		callback(null, fileDescriptor);
+	}, 1);
+};
+
+/**
+ * @param {string|Buffer|URL} path path to file
+ * @param {string} [flags='r'] file system access flags
+ * @param {integer} [mode=0o666] file mode to use when creating file
+ * @returns {integer}
+ */
+fs.openSync = (path, flags = 'r', mode = 0o666) => {
+	const tiFile = getTiFileFromPathLikeValue(path);
+	if (!tiFile.exists()) {
+		// TODO: Support creating file with specific mode
+		if (!tiFile.createFile()) {
+			// Oh crap, we failed to create the file. why?
+			if (!tiFile.parent.exists()) {
+				// parent does not exist!
+				throw noSuchFile('open', path);
+			}
+
+			throw new Error(`failed to create file at path ${path}`);
+		}
+	} else if (flags) {
+		// file/dir exists...
+		if ((flags.charAt(0) === 'w' || flags.charAt(0) === 'a') && tiFile.isDirectory()) {
+			// If user is trying to write or append and it's a directory, fail
+			throw new illegalOperationOnADirectory('open', path);
+		}
+		if (flags.length > 1 && flags.charAt(1) === 'x') {
+			// If user has "exclusive" flag on, fail if file already exists
+			throw new fileAlreadyExists('open', path);
+		}
+	}
+	const tiMode = FLAGS_TO_TI_MODE.get(flags);
+	// TODO: If invalid flag, throw an error!
+	return createFileDescriptor(path, tiFile.open(tiMode));
+};
+
+/**
+ * @callback readCallback
+ * @param {Error} err - Error if one occurred
+ * @param {integer} bytesRead - number of bytes read
+ * @param {Buffer} buffer buffer
+ */
+
+/**
+ * @param {integer} fd file descriptor
+ * @param {Buffer} buffer buffer to read into
+ * @param {integer} offset the offset in the buffer to start writing at.
+ * @param {integer} length integer specifying the number of bytes to read.
+ * @param {integer} position where to begin reading from in the file
+ * @param {readCallback} callback async callback
+ */
+fs.read = (fd, buffer, offset, length, position, callback) => {
+	const tiFileStream = streamForDescriptor(fd);
+	tiFileStream.read(buffer, offset, length, readObj => {
+		if (!readObj.success) {
+			callback(new Error(readObj.error));
+			return;
+		}
+		callback(null, readObj.bytesProcessed, buffer);
+	});
+};
+
+/**
+ * @param {integer} fd file descriptor
+ * @param {Buffer} buffer buffer to read into
+ * @param {integer} offset the offset in the buffer to start writing at.
+ * @param {integer} length integer specifying the number of bytes to read.
+ * @param {integer} position where to begin reading from in the file
+ * @returns {integer} bytes read
+ */
+fs.readSync = (fd, buffer, offset, length, position) => {
```

Comment:
<!--
  0 failure: 
  1 warning:  common/Resources/...
  
  
  DangerID: danger-id-Danger;
  File: common/Resources/ti.internal/extensions/node/fs.js;
  Line: 656;
-->

- :warning: common/Resources/ti.internal/extensions/node/fs.js line 656 – 'position' is defined but never used. Allowed unused args must match /^_.+/. (no-unused-vars)